lib/ipmi_backend_utils.pm

```diff
@@ -129,11 +129,17 @@ sub setup_console_in_grub {
             die "Host Hypervisor is not xen or kvm";
         }
 
+        #enable Intel VT-d for SR-IOV test running on intel SUTs
+        my $intel_option = "";
+        if (get_var("ENABLE_SRIOV_NETWORK_CARD_PCI_PASSSHTROUGH") && script_run("grep Intel /proc/cpuinfo") == 0) {
+            $intel_option = "intel_iommu=on";
+        }
+
         $cmd
           = "cp $grub_cfg_file ${grub_cfg_file}.org "
           . "\&\& sed -ri '/($bootmethod\\s*.*$search_pattern)/ "
           . "{s/(console|loglevel|log_lvl|guest_loglvl)=[^ ]*//g; "
-          . "/$bootmethod\\s*.*$search_pattern/ s/\$/ console=$ipmi_console,115200 console=tty loglevel=5/;}; "
+          . "/$bootmethod\\s*.*$search_pattern/ s/\$/ console=$ipmi_console,115200 console=tty loglevel=5 $intel_option/;}; "
```

Comment:
intel_iommu is a general option ? Can be enabled on any host with intel ? I think you should detect vt-d flag on intel cpu. I think intel_iommu is only available with intel vt-d extension.

Reply:
Our virtualization test machines should support vt-d, but it is really safer to detect vt-d flag than just Intel cpu. Thanks.

Reply:
Not found vt-d in cpu flag, weird. Just keep using 'Intel' as the flag. I'll keep an eye on this feature.

Reply:
Maybe it is not listed by lscpu. It is somewhere else.

Reply:
After double check  I think detecting whether iommu is really activated is really important, so you can stop the test asap if iommu is not activated even with intel_iommu=on. You can do like this:
     AMD-based machines:

     dmesg | grep AMD-Vi
     ...
     AMD-Vi: Enabling IOMMU at 0000:00:00.2 cap 0x40
     AMD-Vi: Lazy IO/TLB flushing enabled
     AMD-Vi: Initialized for Passthrough Mode
     ...

    Intel-based machines:

     dmesg | grep -e DMAR -e IOMMU
     ...
     DMAR:DRHD base: 0x000000feb03000 flags: 0x0
     IOMMU feb03000: ver 1:0 cap c9008020e30260 ecap 1000
     ...

Reply:
Additionally make sure your $intel_option does no append to xen kernel

Reply:
Thanks, Wayne. I am investigating as well. I'll make this check in my tests, including your investigation and my further discovers.